frontend/integration-tests/tests/kubevirt/vm.overview.scenario.ts

```diff
@@ -0,0 +1,141 @@
+    expect(vmView.detailViewOS.getText()).toEqual(basicVmConfig.operatingSystem);
+    expect(vmView.detailViewWLP.getText()).toEqual(basicVmConfig.workloadProfile);
+    expect(vmView.detailViewTemplate.getText()).toEqual('openshift/rhel7-generic-small');
+    expect(vmView.detailViewNS.getText()).toEqual(testName);
+    expect(vmView.bootOrder.getText()).toEqual(['rootdisk', 'nic0', 'cloudinitdisk']);
+    expect(vmView.detailViewFlavor.getText()).toEqual(basicVmConfig.flavor);
+    expect(vmView.detailViewFlavorDes.getText()).toEqual('1 CPU, 2G Memory');
+
+    // Empty fields when vm is off
+    expect(vmView.detailViewIP.getText()).toEqual('---');
+    expect(vmView.detailViewPod.getText()).toEqual('---');
+    expect(vmView.detailViewFQDN.getText()).toEqual('---');
+    expect(vmView.detailViewNode.getText()).toEqual('---');
+
+    // Edit button is enabled when VM is off
+    expect(vmView.detailViewEditBtn.isEnabled()).toBe(true);
+  });
+
+  it('Check VM details in overview when vm is running', async() => {
+    await vm.action('Start');
+    expect(vmView.statusIcon(vmView.statusIcons.running).isPresent()).toBeTruthy();
+
+    // Empty fields turn into non-empty
+    expect(vmView.detailViewIP.getText()).toContain('10');
+    // Known issue for FQDN: https://bugzilla.redhat.com/show_bug.cgi?id=1688124
+    expect(vmView.detailViewFQDN.getText()).toEqual(vmName);
+    expect(vmView.detailViewPod.$('a').getText()).toContain('virt-launcher');
+    expect(vmView.detailViewNode.$('a').getText()).not.toEqual('---');
+
+    // Edit button is disabled when VM is running
+    expect(vmView.detailViewEditBtn.isEnabled()).toBe(false);
+  });
+
+  it('Edit VM flavor', async() => {
+    const newVMDescription = 'edited vm description';
+    await vm.action('Stop');
+    await isLoaded();
+
+    // Cancel edit
+    await vmView.detailViewEditBtn.click();
+    await fillInput(vmView.detailViewDesTextarea, newVMDescription);
+    await vm.selectFlavor('Custom');
+    await fillInput(vmView.detailViewFlavorCPU, '2');
+    await fillInput(vmView.detailViewFlavorMemory, '4');
+    await vmView.detailViewCancelBtn.click();
+    expect(vmView.detailViewDescription.getText()).toEqual(testName);
+    expect(vmView.detailViewFlavorDes.getText()).toEqual('1 CPU, 2G Memory');
+
+    // Save edit
+    await vmView.detailViewEditBtn.click();
+    await fillInput(vmView.detailViewDesTextarea, newVMDescription);
+    await vm.selectFlavor('Custom');
+    await fillInput(vmView.detailViewFlavorCPU, '2');
+    await fillInput(vmView.detailViewFlavorMemory, '4');
+    await vmView.detailViewSaveBtn.click();
+    await isLoaded();
+    expect(vmView.detailViewDescription.getText()).toEqual(newVMDescription);
+    expect(vmView.detailViewFlavorDes.getText()).toEqual('2 CPU, 4G Memory');
+  });
+
+  it('Expose some VM services', async() => {
+    expect($(`[href="/k8s/ns/${testName}/services/${vmName}-service-ssh"]`).getText()).toEqual(`${vmName}-service-ssh`);
```

Comment:
Nicely solved selector. Could you create a function that generates the service selector similarly to what @suomiy suggests in the comment below in virtualMachine.view.ts?

Reply:
Thanks, updated it accordingly.